app-engine/java/src/com/google/cloud/tools/intellij/appengine/java/cloud/AppEngineModuleDeploymentSourceProvider.java

```diff
@@ -61,18 +61,18 @@
       if (facetManager.getFacetByType(AppEngineStandardFacetType.ID) != null
           || facetManager.getFacetByType(AppEngineFlexibleFacetType.ID) != null) {
         AppEngineEnvironment environment =
-            projectService
-                .getModuleAppEngineEnvironment(module)
-                .orElseThrow(() -> new RuntimeException("No environment."));
+            projectService.getModuleAppEngineEnvironment(module).orElse(null);
 
-        if (ModuleType.is(module, JavaModuleType.getModuleType())
-            && projectService.isJarOrWarMavenBuild(module)) {
-          moduleDeploymentSources.add(
-              createMavenBuildDeploymentSource(project, module, environment));
-        }
+        if (environment != null) {
```

Comment:
just fyi I used a null check here instead of an `ifPresent` lambda due to the `hasStandardModules` side effect